guardian/grid
# 1194 interim filters

## What does this change?
The aim of this new functionality is to make it easier for Users to find and select images that have usage rights appropriate to their needs and to provide visual indicators regarding the usage rights of an image and its suitablility for a particular use.

**Addition of new filter control**
Change amens the kahuna UI to provide a two tier top bar and re-configures the arrangement of controls across the two bars to provide more screen space for menu items. It also introduces a 'filter' dropdown option. This is a configurable control that allows for the easy construction of frequently used chip searches providing them with meaningful names. It also introduces a drop down control for the sort order options (no new ordering behaviour is added but this will be extended in future).

The metadata info panel has also been extended to include additional usage rights information and icons in a configurable fashion. This too is controlled through configuration.

The new configuration options for kahuna are;

        |usePermissionsFilter=true
        |permissionsDefault="allPermissions"
        |permissionsOptions=`[{"id":"allPermissions","label":"All permissions","mapping":"","payable":"none"},{"id":"usableForNews","label":"Usable for News","mapping":"category:agency","payable":"false"}]`

Note the permissionsOptions are configured as an array of JSON objects of the form;

{
  "id": "exampleId",  <-- internally used identifier
  "label": "Example Label", <-- the label that will be displayed in the drop down list
  "mapping": "category:agency,source:example", <-- a csv list of the required search chips
  "payable": "none|true|false" <-- whether 'include payable images' should be forced to true, false or left unchanged (none)
}

Filter option mappings can include more than a single filter 'chip' - when multiple chips are required they should be a comma separated list.

All the permissions options can be changed as required. Switching usePermissionsFilter and usageRightsSummary to 'false' will leave the UI unchanged from its current behaviour.

If all the configuration settings are excluded (ie no change to current config) the UI will continue to operate in an unaltered fashion (ie once the code is merged there is no requirement to make any modifications to the config unless you want to utilise the functionality).

The new UI appears as;

![Screenshot 2023-12-14 at 16 48 30](https://github.com/guardian/grid/assets/128470622/a1d82bfd-9581-4f3f-b37a-1f1a19edf74f)

**Additional Usage Right Summary Info**
This change introduces a new control that appears in the 'Rights and restrictions' section of the Info panel. Thie control replicates the existing wording bu then in certain circumstances adds an icon and additional text into the section to provide more guidance to the user regaridn the rights of an image. The icon, text  and display rules for this addditional information are set in an organisation specific file in the kahuna/.../gr-usagerights-summay folder - e.g. gr-usagerights-bbc.tsx. 

This feature is turned on and off with a kahuna config flag;

        |usageRightsSummary=true

If this config is excluded the UI will continue to work unchanged.

The usage rights summary outputs as configured by the BBC logic embedded within this PR are;

Usable for all images (green);
![Screenshot 2023-12-18 at 12 55 02](https://github.com/guardian/grid/assets/128470622/e54233c1-bf73-48dd-b0a1-c55c2a6da662)

Usable for news (amber);
![Screenshot 2023-12-18 at 12 55 31](https://github.com/guardian/grid/assets/128470622/59433f02-5df3-4b72-9930-edbcc92b9f92)

Payable Images (red);
![Screenshot 2023-12-18 at 12 56 55](https://github.com/guardian/grid/assets/128470622/8a54ec22-ab42-41ff-8d19-f75bc5be6ff0)

Any icons text or logic can be configured in an appropriate tsx file.

## How should a reviewer test this change?

Tester should ensure that UI behaves unchanged with config options switched to false.

User should ensure that with options switched to true all new controls appear and function as required providing correct filter URLs to the search endpoint.

## Tested? Documented?
